src/server.c

```diff
@@ -4857,6 +4857,18 @@ void infoCommand(client *c) {
 }
 
 void monitorCommand(client *c) {
+    if ((c->flags & CLIENT_DENY_BLOCKING) && !(c->flags & CLIENT_MULTI)) {
+        /**
+         * A Client that has CLIENT_DENY_BLOCKING flag on
+         * expect a reply per command and so can not execute monitor.
+         *
+         * Notice that we have a special treatment for multi because of
+         * backword compatibility
+         */
+        addReplyError(c, "monitor is not allow on DENY BLOCKING client");
+        return;
```

Comment:
i think we discussed this background compatibility (allowing MULTI) for SUBSCRIBE, not for MONITOR.
am i missing anything?
did you forget to put it in SUBSCRIBE?
do you see a reason to have it in MONITOR too?

Reply:
@itamarhaber do you see any reason anyone will execute MONITOR inside MULTI?
i guess SUBSCRIBE makes some sense, but for MONITOR i think we can break it if we want, WDYT?

Reply:
I don't see how one would use MONITOR, SUBSCRIBE, or any client-blocking command for that matter in a MULTI block, even if it is currently allowed.

Reply:
Also, notice that currently monitor inside multi will mess the multi reply:
```
>> telnet localhost 6379
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
multi
+OK
monitor
+QUEUED
set x 1
+QUEUED
set y 2
+QUEUED
exec
*3
+OK
+1604748860.711977 [0 127.0.0.1:59260] "set" "x" "1"
+OK
+1604748860.712036 [0 127.0.0.1:59260] "set" "y" "2"
+OK
+1604748860.712056 [0 127.0.0.1:59260] "exec"
```

FYI I am using telnet because the redis-cli, after sending the monitor, will wait for messages and not allow completing the transaction (which means that we will also not get any monitor messages). If we will not allow those commands inside multi then we have no issue on redis-cli, otherwise, we might want to consider fixing the redis-cli?

Reply:
I don't care too much about fixing redis-cli for this edge case, i think we can breah/fix this for MONITOR (don't allow inside MULTI).

And i think we need to keep allowing SUBSCRIBE inside MULTI, (someone may try to use it with redis.py pipeline and get a transaction by mistake.

p.s. i noticed you forgot to handle PSUBSCRIBE (you need to copy your patch from subscribeCommand)